pkg/scheduler/framework/runtime/framework.go

```diff
@@ -384,6 +395,91 @@ func NewFramework(r Registry, profile *config.KubeSchedulerProfile, opts ...Opti
 	return f, nil
 }
 
+func getScoreWeights(f *frameworkImpl, totalPriority int64, plugins []config.Plugin) (error, int64) {
+	for _, e := range plugins {
+		// a weight of zero is not permitted, plugins can be disabled explicitly
+		// when configured.
+		f.scorePluginWeight[e.Name] = int(e.Weight)
+		if f.scorePluginWeight[e.Name] == 0 {
+			f.scorePluginWeight[e.Name] = 1
+		}
+
+		// Checks totalPriority against MaxTotalScore to avoid overflow
+		if int64(f.scorePluginWeight[e.Name])*framework.MaxNodeScore > framework.MaxTotalScore-totalPriority {
+			return fmt.Errorf("total score of Score plugins could overflow"), 0
+		}
+		totalPriority += int64(f.scorePluginWeight[e.Name]) * framework.MaxNodeScore
+	}
+	return nil, totalPriority
+}
+
+func (f *frameworkImpl) expandMultiPointPlugins(profile *config.KubeSchedulerProfile, pluginsMap map[string]framework.Plugin) error {
+	// initialize MultiPoint plugins
+	for _, e := range f.getExtensionPoints(profile.Plugins) {
+		plugins := reflect.ValueOf(e.slicePtr).Elem()
+		pluginType := plugins.Type().Elem()
+		// build enabledSet of plugins already registered via normal extension points
+		// to check double registration
+		enabledSet := sets.NewString()
+		for _, plugin := range e.plugins.Enabled {
+			enabledSet.Insert(plugin.Name)
+		}
+
+		disabledSet := sets.NewString()
+		for _, disabledPlugin := range e.plugins.Disabled {
+			disabledSet.Insert(disabledPlugin.Name)
+		}
```

Comment:
I suppose we're not covering the logic properly. See my another comment, the disabledSet seems to have dropped default pluginSet's `disabled`, and thus the data here is incorrect.

Reply:
I commented on that above, but we don't have any default plugins that are disabled. I can re-add [the logic](https://github.com/kubernetes/kubernetes/commit/8a81833da8c9a48b65ced41a00ad8289cccf7f63#) to handle them just in case we add any in the future.

If we do that, this will then handle those cases as well. That's because the default plugins and custom config are merged at this point into a single config.

Reply:
@damemi  we shouldn't make assumptions about the versioned logic. The implementation here should handle potential changes in future versions.

Reply:
Re-added the logic to handle default disabled plugins

Reply:
Yeap, we shouldn't make the assumptions as other consumers like ClusterAutoscaler may instantiate the plugins any way they like.